### PR TITLE
ENT-351 Add tpa_hint to urls in SAP course export

### PR DESCRIPTION
**Pull request by brittneyexline**
**Description:** We weren't adding the tpa_hint of the enterprise customer identity provider to the launchURL's of the courses, which we need in order to take advantage of all of our SSO logic.

**JIRA:** https://openedx.atlassian.net/browse/ENT-351

**Dependencies:** Soft dependency on https://github.com/edx/edx-enterprise/pull/94 for sandbox testing purposes - so you will notice the code is included here, but I can separate those changes out once this is approved.

**Merge deadline:** 5/5/2017

**Installation instructions:** Requires an environment with the catalog service running.

**Testing instructions:**

Edxapp sandbox: https://brittneyexline.sandbox.edx.org

1. Set up a configuration for an enterprise customer and success factors against our test SAP instance. (This has been done already on the sandbox)
2. Make a catalog at least one course. (On the sandbox it is "One Catalog to rule them all")
3. Run the catalog export script, ensure that it has run successfully. To do this on the sandbox, ssh to the edxapp sandbox. Then run the following commands:
$ sudo -H -u edxapp bash
$ source /edx/app/edxapp/edxapp_env
$ cd /edx/app/edxapp/edx-platform/
$ /edx/bin/python.edxapp /edx/bin/manage.edxapp lms transmit_courseware_data --catalog_user staff --settings=aws

To see the courses on SAP, follow the instructions here for Importing Courses to view which courses have been sent. Verify that the launchURL that was sent has the appropriate tpa_hint as a query parameter.
https://openedx.atlassian.net/wiki/pages/viewpage.action?spaceKey=SOL&title=SAP+Testing+and+Configuration+notes

**Merge checklist:**

- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: integrated_channels/sap_success_factors/utils.py [hunk 1]
```diff
@@ -276,8 +276,12 @@ def get_course_track_selection_url(enterprise_customer, course_id):
     """
     netloc = enterprise_customer.site.domain
     scheme = COURSE_URL_SCHEME
+    if enterprise_customer.identity_provider:
+        tpa_hint = 'tpa_hint={}'.format(enterprise_customer.identity_provider)
+    else:
+        tpa_hint = ''
```
Comment (douglashall) [inline integrated_channels/sap_success_factors/utils.py]: Nit: Define tpa_hint with default value first.
```
tpa_hint = ''
idp = enterprise_customer.identity_provider
if idp:
    tpa_hint = 'tpa_hint={}'.format(idp)
```

**Comment by bradenmacdonald** [comment]
FYI, I just [suggested here](https://openedx.atlassian.net/browse/ENT-361?focusedCommentId=258349&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-258349) that we may be outgrowing `tpa_hint`, so we may want to consider moving toward an enterprise-specific parameter in the future rather than overloading tpa_hint. Doesn't need to affect this PR and others may disagree but it's worth keeping in mind.

**Comment by brittneyexline** [comment]
Thanks @bradenmacdonald - it's a good point. I think once we have a landing page that is specific to each enterprise (which is in development in the next couple weeks), we will be switching this out to use that url.

File: integrated_channels/sap_success_factors/utils.py [hunk 2]
```diff
@@ -285,8 +285,12 @@ def get_course_track_selection_url(enterprise_customer, course_id):
     """
     netloc = enterprise_customer.site.domain
     scheme = COURSE_URL_SCHEME
+    if enterprise_customer.identity_provider:
+        tpa_hint = 'tpa_hint={}'.format(enterprise_customer.identity_provider)
```
Comment (haikuginger) [inline integrated_channels/sap_success_factors/utils.py]: Can we use `urlencode` to ensure we're following best practices here?

**Pull request merged**